### DISCUSSION TITLE
Is there an easy way to copy last command from terminal to clipboard?

### DISCUSSION THREAD

**Discussion by theRoboxx** [leo-arch/clifm#214] [Q&A] [answered]
I'm looking for an alternative way to copy a given command from the terminal.

My current steps for reproduction are the following:

1. Use the arrow keys to switch to the command
2. Select text with doubleclick + mouse
![swappy-20230411_130216](https://user-images.githubusercontent.com/63428105/231149728-7574b8f3-24d0-4ae4-9990-40731f3af84a.png)

4. copy with `ctrl+shift+c`

Would it be possible to implement a shortcut that handle step 2+3?

**Comment by leo-arch** [comment]
Hi @theRoboxx. I normally do it like this:

1. Select text with mouse/trackpad.
2. Paste it by middle clicking.

**Reply by leo-arch** [reply, marked as answer]
If you want to copy the line buffer to the clipboard, to later retrieve it via <kbd>Ctrl-v</kbd> or the `Paste` command in a context menu, you can do it as follows:

NOTE: You need the latest version of **clifm**, so that you need to clone it and build it as always.

1. Create a new plugin, say `xclip.sh`, with the following content and drop it into the plugins directory (`~/.config/clifm/plugins`):

```sh
#!/bin/sh

if [ -z "$CLIFM_LINE" ]; then
    printf "clifm: Empty line buffer\n" 2>&1
    exit 1
fi

printf "%s" "$CLIFM_LINE" | xclip -selection clipboard
printf "clifm: Line buffer copied into the clipboard\n"
```
NOTE: Replace `xclip` by your preferred X clipboard application.

2.  Bind this plugin to an action (via the `actions edit` command) by adding this line:
```
plugin1=xclip.sh
```
3. Bind a keyboard shortcut (here we use <kbd>Ctrl-y</kbd>) to the newly created action. Run `kb edit` and add this line:
```
plugin1=\C-y
```
4. Restart **clifm**

That's it. Now you can copy the current line buffer to the clipboard by just pressing <kbd>Ctrl-y</kbd>.

**Reply by leo-arch** [reply]
An improved version of the plugin is already in the main repo: [xclip.sh](https://github.com/leo-arch/clifm/blob/master/plugins/xclip.sh).

**Reply by theRoboxx** [reply]
Thanks a lot, I will test it later. :+1:
